modules/ravestate_roboyio/__init__.py

```diff
@@ -36,11 +40,22 @@
             msg_type=RecognizedSpeech,
             always_signal_changed=True)
 
-        @state(read=recognized_speech.id())
+        #service_recognized_speech = Ros2CallProperty(name="service_recognized_speech",
+        #                                             service_name="/roboy/cognition/speech/recognition",
+        #                                             service_type=RecognizeSpeech,
+        #                                             call_timeout=10.0)
+
+        @state(cond=startup(), read="interloc:all") #, read=service_recognized_speech.id(), write=service_recognized_speech.id())
         def roboy_input(ctx: ContextWrapper):
-            handle_single_interlocutor_input(ctx, ctx[recognized_speech.id()].text)
+            while not ctx.shutting_down():
+                #ctx[service_recognized_speech.id()] = RecognizeSpeech.Request()
+                #result = ctx[service_recognized_speech.id()]
+                result = listen()
+                if result:  # TODO what is returned when nothing understood?
+                    handle_single_interlocutor_input(ctx, result)
```

Comment:
listen() from pyroboy worked, we should remove the ros2-property implementation

Reply:
👍